### PR TITLE
New Adprime bidder adapter

### PR DESCRIPTION
**Pull request by Adprime**
## Type of change

- [x] New bidder adapter

- test parameters for validating bids
```
 {
       bidder: 'adprime',
                            params: {
                                placementId: 0,
                                traffic: 'banner'
                            }
                        }
```

- contact email of the adapter’s maintainer
```
rafal@adprime.com
```
- [x] official adapter submission

## Other information
Docs: https://github.com/prebid/prebid.github.io/pull/1952

### CODE REVIEW

File: modules/adprimeBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    const len = validBidRequests.length;
+
+    for (let i = 0; i < len; i++) {
+      let bid = validBidRequests[i];
+      let traff = bid.params.traffic || BANNER
+
+      placements.push({
+        placementId: bid.params.placementId,
+        bidId: bid.bidId,
+        sizes: bid.mediaTypes && bid.mediaTypes[traff] && bid.mediaTypes[traff].sizes ? bid.mediaTypes[traff].sizes : [],
```
Comment (msm0504) [inline modules/adprimeBidAdapter.js:71]: `sizes` will only have the size information for banner media type. The structures for video and native are different.

See the media type [examples](http://prebid.org/dev-docs/publisher-api-reference.html#addAdUnits-MediaTypes)

File: modules/adprimeBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    for (let i = 0; i < serverResponse.body.length; i++) {
+      let resItem = serverResponse.body[i];
+      if (isBidResponseValid(resItem)) {
+        response.push(resItem);
+      }
+    }
+    return response;
+  },
+
+  getUserSyncs: (syncOptions, serverResponses) => {
```
Comment (msm0504) [inline modules/adprimeBidAdapter.js:96]: Is this empty function needed?

File: modules/adprimeBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import { BANNER, NATIVE, VIDEO } from '../src/mediaTypes.js';
+import * as utils from '../src/utils.js';
+
+const BIDDER_CODE = 'adprime';
+const AD_URL = 'https://delta.adprime.com/?c=o&m=multi';
```
Comment (msm0504) [inline modules/adprimeBidAdapter.js:6]: Responses from your server should be gzipped.

File: modules/adprimeBidAdapter.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    let request = {
+      'deviceWidth': winTop.screen.width,
+      'deviceHeight': winTop.screen.height,
+      'language': (navigator && navigator.language) ? navigator.language.split('-')[0] : '',
+      'secure': 1,
+      'host': location.host,
+      'page': location.pathname,
+      'placements': placements
+    };
+    request.language.indexOf('-') != -1 && (request.language = request.language.split('-')[0])
```
[the remark this answers is not in the document]
Comment (Adprime) [reply modules/adprimeBidAdapter.js:53]: This prevents from receiving language like en-US

File: modules/adprimeBidAdapter.js (hunk 2 above)
Comment (Adprime) [reply modules/adprimeBidAdapter.js:96]: It will be filled after we spin up server for cookiesync

File: modules/adprimeBidAdapter.js (hunk 3 above)
Comment (Adprime) [reply modules/adprimeBidAdapter.js:6]: This one done

File: modules/adprimeBidAdapter.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (Adprime) [reply modules/adprimeBidAdapter.js:53]: O,yes, i understood.I will remove it

File: modules/adprimeBidAdapter.js (hunk 2 above)
Comment (Adprime) [reply modules/adprimeBidAdapter.js:96]: Will remove this completely before cookiesync server will be ready

**Comment by Adprime** [comment]
@msm0504 Thanks for review, i've done all possible changes according to your  comments. There is some issue in circle ci not related to this bidder adapter

**Comment by msm0504** [comment]
@Adprime the code changes look good. However, test coverage is a little low now. Can you add unit tests where the `bid` has media types and where `bidderRequest` has consent objects?

**Comment by Adprime** [comment]
@msm0504 Thanks for review. We have expanded the scope of unit tests.

**Pull request merged**